Source/VMware.vSphereDSC/Tests/Unit/VMHostVssSecurity.Unit.Tests.ps1

```diff
@@ -0,0 +1,628 @@
+                $result | Should -Be $false
+            }
+        }
+
+        Context 'Present, VSS exists and security with the same properties' {
+            BeforeAll {
+                $viserverMock = [scriptblock]::Create($ExecutionContext.InvokeCommand.ExpandString($script:viServerCode))
+                $vmHostMock = [scriptblock]::Create($ExecutionContext.InvokeCommand.ExpandString($script:vmHostCode))
+                $networkSystemMock = [scriptblock]::Create($ExecutionContext.InvokeCommand.ExpandString($script:networkSystemSameVssSecurityCode))
+            }
+
+            # Arrange
+            Mock -CommandName Connect-VIServer -MockWith $viserverMock -ModuleName $script:moduleName
+            Mock -CommandName Get-VMHost -MockWith $vmHostMock -ModuleName $script:moduleName
+            Mock -CommandName Get-View -MockWith $networkSystemMock -ModuleName $script:moduleName
+
+            $resource = New-Object -TypeName $script:resourceName -Property $script:resourceProperties
+
+            It 'Should return $true (The desired VSS is configured correctly)' {
+                # Act
+                $result = $resource.Test()
+
+                # Assert
+                $result | Should -Be $true
+            }
+        }
+
+        Context 'Absent and VSS does not exist' {
+            BeforeAll {
+                $viserverMock = [scriptblock]::Create($ExecutionContext.InvokeCommand.ExpandString($script:viServerCode))
+                $vmHostMock = [scriptblock]::Create($ExecutionContext.InvokeCommand.ExpandString($script:vmHostCode))
+                $networkSystemMock = [scriptblock]::Create($ExecutionContext.InvokeCommand.ExpandString($script:networkSystemNoVssSecurityCode))
+            }
+
+            # Arrange
+            Mock -CommandName Connect-VIServer -MockWith $viserverMock -ModuleName $script:moduleName
+            Mock -CommandName Get-VMHost -MockWith $vmHostMock -ModuleName $script:moduleName
+            Mock -CommandName Get-View -MockWith $networkSystemMock -ModuleName $script:moduleName
+
+            $resource = New-Object -TypeName $script:resourceName -Property $script:resourcePropertiesAbsent
+
+            It 'Should call the Connect-VIServer mock with the passed server and credentials once' {
+                # Act
+                $resource.Test()
+
+                # Assert
+                $assertMockCalledParams = @{
+                    CommandName = 'Connect-VIServer'
+                    ParameterFilter = { $Server -eq $script:resourceProperties.Server -and $Credential -eq $script:resourceProperties.Credential }
+                    ModuleName = $script:moduleName
+                    Exactly = $true
+                    Times = 1
+                    Scope = 'It'
+                }
+                Assert-MockCalled @assertMockCalledParams
+            }
+
+            It 'Should call Get-VMHost with the passed server and name once' {
+                # Act
+                $resource.Get()
```

Comment:
Here it should be $resource.Test() and the same for the below test.

Reply:
Changed